plugins/urls/__init__.py

```diff
@@ -175,113 +238,74 @@ def message_handler(self, event=MessageReceived):
 
-                    if domain is not None and "/" in domain:
-                        domain = domain.split("/")[0]
-                    if domain is None:
-                        source.respond(title)
-                    else:
-                        source.respond("\"%s\" at %s" % (title, domain))
-                else:
-                    self.logger.warn(_("Unknown target type: %s [%s]")
-                                     % (target, target.__class__))
+            try:
+                if _port:
+                    _port = int(_port)
+            except ValueError:
+                self.logger.warn("Invalid port: {0}", _port)
+                continue
+
+            _translated = self.translate_prefix(_prefix)
+
+            if len(_path) > 0 and len(_translated) > 0:
+                # Remove translated prefix chars.
+                for char in reversed(_translated):
+                    if _path[-1] == char:
+                        _path = _path[:-1]
+            elif len(_domain) > 0 and len(_translated) > 0:
+                # Remove translated prefix chars.
+                for char in reversed(_translated):
+                    if _domain[-1] == char:
+                        _domain = _domain[:-1]
+
+            _url = URL(self, _protocol, _basic, _domain, _port, _path)
+
+            self.channels.get(protocol.name, {}) \
+                .get(source.name, {})["last"] = _url.text
+
+            yield self.run_handlers(_url, {
+                "event": event,
+                "config": self.config
+            })
 
     def urls_command(self, protocol, caller, source, command, raw_args,
-                     parsed_args):
+                     args):
         """
         Command handler for the urls command
         """
 
-        args = raw_args.split()  # Quick fix for new command handler signature
         if not isinstance(source, Channel):
-            caller.respond(__("This command can only be used in a channel."))
+            caller.respond("This command can only be used in a channel.")
             return
         if len(args) < 2:
-            caller.respond(__("Usage: {CHARS}urls <setting> <value>"))
-            caller.respond(__("Operations: set <on/off> - Enable or disable "
-                              "title parsing for the current channel"))
-            caller.respond("            %s" % __("shortener <name> - Set "
-                                                 "which URL shortener to use "
-                                                 "for the current channel"))
-            caller.respond("            %s" % __("Shorteners: %s")
-                           % ", ".join(self.shorteners.keys()))
+            caller.respond("Usage: {CHARS}%s <setting> <value>" % command)
+            caller.respond("Operations: set <on/off> - Enable or disable "
+                           "title parsing for the current channel")
+            caller.respond("            %s" % "shortener <name> - Set "
```

Comment:
Double %s would've been an artifact of the old plugin doing things with translations I think.

This code was basically copied from there, since it already worked. Refactor might not be a bad idea.